core/src/com/unciv/logic/multiplayer/DropBox.kt

```diff
@@ -0,0 +1,169 @@
+package com.unciv.logic.multiplayer
+
+import com.unciv.logic.GameSaver
+import com.unciv.ui.utils.UncivDateFormat.parseDate
+import java.io.*
+import java.net.HttpURLConnection
+import java.net.URL
+import java.nio.charset.Charset
+import java.util.*
+import kotlin.collections.ArrayList
+
+
+object DropBox {
+    fun dropboxApi(url: String, data: String = "", contentType: String = "", dropboxApiArg: String = ""): InputStream? {
+
+        with(URL(url).openConnection() as HttpURLConnection) {
+            requestMethod = "POST"  // default is GET
+
+            @Suppress("SpellCheckingInspection")
+            setRequestProperty("Authorization", "Bearer LTdBbopPUQ0AAAAAAAACxh4_Qd1eVMM7IBK3ULV3BgxzWZDMfhmgFbuUNF_rXQWb")
+
+            if (dropboxApiArg != "") setRequestProperty("Dropbox-API-Arg", dropboxApiArg)
+            if (contentType != "") setRequestProperty("Content-Type", contentType)
+
+            doOutput = true
+
+            try {
+                if (data != "") {
+                    // StandardCharsets.UTF_8 requires API 19
+                    val postData: ByteArray = data.toByteArray(Charset.forName("UTF-8"))
+                    val outputStream = DataOutputStream(outputStream)
+                    outputStream.write(postData)
+                    outputStream.flush()
+                }
+
+                return inputStream
+            } catch (ex: Exception) {
+                println(ex.message)
+                val reader = BufferedReader(InputStreamReader(errorStream))
+                val responseString = reader.readText()
+                println(responseString)
+
+                // Throw Exceptions based on the HTTP response from dropbox
+                if (responseString.contains("path/not_found/"))
+                    throw FileNotFoundException()
+                if (responseString.contains("path/conflict/file"))
+                    throw FileStorageConflictException()
```

Comment:
Marking everything that changed
- Changed DropBoxFileConflictException to FileStorageConflictException to match FileStorage scheme